client/src/components/SessionForm.jsx

```diff
@@ -76,13 +91,94 @@ function SessionForm(props) {
       timeout: timeout,
     };
     console.log(sessionData);
-    await addSession(sessionData);
+
+    if (isEdit) {
+      await editSession(session._id, sessionData);
+      setSession({ ...sessionData, _id: session._id });
+    } else {
+      const updatedSession = await addSession(sessionData);
+      console.log(updatedSession);
+      setSessions([...sessions, updatedSession.data.result]);
+    }
+  };
+
+  const formSetup = () => {
+    const splitClass = session.class.split(/([0-9]+)/);
+    setCourseCode(splitClass[0]);
+    setCourseNumber(splitClass[1]);
+    if (splitClass.length > 2) {
+      setCourseSuffix(splitClass[2]);
+    }
+    setLocation(session.location);
+    setAttendees(
+      session.attendees.length > 0
+        ? session.attendees.join(',')
+        : session.attendees,
+    );
+    setNotes(session.notes);
+
+    setIsLater(!session.active);
+    const startSeconds = session.startTime;
+
+    if (!session.active) {
+      const startDate = new Date(startSeconds * 1000);
+      const year = startDate.getFullYear();
+      const month = startDate.getMonth();
+      const day = startDate.getDate();
+
+      const processedMonth = month < 10 ? '0'.concat(month) : ''.concat(month);
+      const processedDay = day < 10 ? '0'.concat(day) : ''.concat(day);
+
+      const processedDate = ''
+        .concat(year)
+        .concat('-')
+        .concat(processedMonth)
+        .concat('-')
+        .concat(processedDay);
+      setDate(processedDate);
+
+      const startHour = startDate.getHours();
+      const startMinute = startDate.getMinutes();
+
+      const processedStartHour =
+        startHour < 10 ? '0'.concat(startHour) : ''.concat(startHour);
+      const processedStartMinute =
+        startMinute < 10 ? '0'.concat(startMinute) : ''.concat(startMinute);
+      const processedStartTime = processedStartHour
+        .concat(':')
+        .concat(processedStartMinute);
+      setStartTime(processedStartTime);
+    }
+
+    const endDate = new Date((startSeconds + session.timeout) * 1000);
+    const endHour = endDate.getHours();
+    const endMinute = endDate.getMinutes();
+
+    const processedEndHour =
+      endHour < 10 ? '0'.concat(endHour) : ''.concat(endHour);
+    const processedEndMinute =
+      endMinute < 10 ? '0'.concat(endMinute) : ''.concat(endMinute);
+    const processedEndTime = processedEndHour
+      .concat(':')
+      .concat(processedEndMinute);
+    setEndTime(processedEndTime);
+
+    const defaultTimeout = 43200;
+
+    if (session.timeout !== defaultTimeout) {
+      setEndTimeDefined(true);
+    }
   };
 
   return (
     <Modal
       onClose={() => setOpen(false)}
-      onOpen={() => setOpen(true)}
+      onOpen={() => {
+        setOpen(true);
+        if (isEdit) {
+          formSetup();
+        }
+      }}
       open={open}
       trigger={button}
     >
```

Comment:
Right now, clicking or even trying to type on the form also opens the Details Modal when you're in edit mode. To fix this, add an `onClick` function to the `<Form>` element where you call stopPropagation (like earlier). (It should be right under these lines, but it won't let me comment on those because nothing was changed since last time)